### PR TITLE
Add new bookmark endpoint

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
This PR adds the API scaffold for Channel Bookmarks and the create channel bookmark endpoint.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-54418

#### Release Note
```release-note
Added create channel bookmark endpoint at /api/v4/channels/{channel_id}/bookmarks.
```

### CODE REVIEW

File: server/channels/api4/channel_bookmark_test.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCreateChannelBookmark(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	t.Run("should not work without a license", func(t *testing.T) {
```
Comment (enahum) [inline server/channels/api4/channel_bookmark_test.go:18]: not sure this is an enterprise feature.

File: server/i18n/en.json [hunk 2]
```diff
@@ -231,6 +231,22 @@
     "id": "api.channel.add_user_to_channel.type.app_error",
     "translation": "Can not add user to this channel type."
   },
+  {
+    "id": "api.channel.bookmark.create_channel_bookmark.forbidden.app_error",
+    "translation": "Failed to create the channel bookmark."
+  },
+  {
+    "id": "api.channel.bookmark.create_channel_bookmark.license.error",
+    "translation": "Your license does not support channel bookmarks."
```
Comment (enahum) [inline server/i18n/en.json:240]: @esethna can you clarify if this is a EE feature? not to my recollection.

**Review by enahum — CHANGES_REQUESTED** [review]

File: server/channels/api4/channel_bookmark.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	}
+	channelBookmark.ChannelId = c.Params.ChannelId
+
+	auditRec := c.MakeAuditRecord("createChannelBookmark", audit.Fail)
+	defer c.LogAuditRec(auditRec)
+	audit.AddEventParameterAuditable(auditRec, "channelBookmark", channelBookmark)
+
+	newChannelBookmark, appErr := c.App.CreateChannelBookmark(c.AppContext, channelBookmark, connectionID)
+	if appErr != nil {
+		c.Err = appErr
```
[the remark this answers is not in the document]
Comment (mgdelacroix) [reply server/channels/api4/channel_bookmark.go:88]: Audit recs are created with an initial status (see last parameter) that then is changed when the request is marked as successful:

```go
auditRec := c.MakeAuditRecord("createChannelBookmark", audit.Fail)
```

File: server/i18n/en.json (hunk 2 above)
Comment (mgdelacroix) [reply server/i18n/en.json:240]: This is based on the prototypes of the system console permissions that reference the feature being enterprise only

Comment (enahum) [reply server/i18n/en.json:240]: ok let's wait for @esethna to confirm, if it is, we should move all the logic to the enterprise repo.

File: server/channels/api4/channel_bookmark_test.go
[the remark this answers is not in the document]
Comment (mgdelacroix) [reply server/channels/api4/channel_bookmark_test.go:1]: Added on 3afac70762, PTAL

**Review by enahum — APPROVED** [review]

**Pull request merged**